### ISSUE TITLE
API for interactive text output and cursor manipulation

### ISSUE THREAD

**Issue by bdw429s** [jline/jline2#159]
What are your thoughts on enhancing the API to give more options for manipulating the cursor?  I am using JLine2 to build a java-based shell and it is working great for the command prompt but, the "commands" my shell executes also want to have the ability to have more interactive behaviors than just outputting text one line at a time.  

An excellent example would be the ability to build a progress bar that would clear and redraw the existing line over and over.  From what I can tell, this isn't possible with the JLine API and projects like https://github.com/brianm/java-progress-bar seem to interact directly with the Ansi classes which I haven't been able to successfully duplicate.  

Does this sort of general-purpose console output functionality fall within the scope of JLine?

**Comment by gnodet** [comment]
I don't really think this should be inside JLine.
JLine uses the jansi library for ansi support, but you can build very complicated things on top of it.
For example, have a look at https://github.com/jledit/jledit

**Comment by bdw429s** [comment]
Thanks for the comment @gnodet.  Perhaps there should be an official project that works with JLine that also supports some of the common functionalities that a CLI would need.  I've found a few places now where something that seemed simple and common in a CLI-based program wasn't possible with JLine2 itself.  (For instance, capturing the next keystroke from the user in a command that says something like "Press q to quit").

Also, for what it's worth, I did get my wget-style progress bar working by directly using org.fusesource.jansi.Ansi and org.fusesource.jansi.AnsiConsole.  The code (CFML) is here:
https://github.com/Ortus-Solutions/commandbox/blob/master/src/cfml/system/util/ProgressBar.cfc#L34

It works pretty well, but I would have hoped to have a slightly easier way of re-drawing the same line of the console over and over.
